examples/kotlin/hello_world/MainActivity.kt

```diff
@@ -48,7 +49,7 @@ class MainActivity : Activity() {
       .addPlatformFilter { DemoFilter() }
       .addPlatformFilter { BufferDemoFilter() }
       .addPlatformFilter { AsyncDemoFilter() }
-      .addStringAccessor("demo-accessor", { "PlatformString" })
+      .addStringAccessor("demo-accessor", { ByteBuffer.wrap("PlatformString".toByteArray(Charsets.UTF_8)) })
```

Comment:
What's fun is that this doesn't catch the existing issue... :(

Reply:
Yeah, this only tests the install path, not the usage path.

When I originally implemented the change I modified the Http::Dispatcher to access the string, and obviously also tested using our internal filters. I failed to do that when I modified to String before :/, which is why I didn't catch the issues I am fixing now.

We could create a filter in Envoy Mobile that tests the example accessors. I will do a subsequent PR for that. We will just not install the filter in our private build.

Reply:
https://github.com/envoyproxy/envoy-mobile/issues/1317!

Reply:
@buildbreaker already have the test code written. Will add in a subsequent PR. Thanks for creating the issue.